fedimint-core/src/module/mod.rs

```diff
@@ -58,6 +59,44 @@ impl TransactionItemAmount {
     };
 }
 
+/// All requests from client to server contain these fields
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ApiRequest<T> {
+    /// Hashed user password if the API requires authentication
+    pub auth: Option<ApiAuth>,
+    /// Parameters required by the API
+    pub params: T,
+}
+
+pub type ApiRequestErased = ApiRequest<JsonValue>;
+
+impl Default for ApiRequestErased {
+    fn default() -> Self {
+        Self {
+            auth: None,
+            params: JsonValue::Null,
+        }
+    }
+}
+
+impl ApiRequestErased {
+    pub fn new<T: Serialize>(params: T) -> ApiRequestErased {
+        Self {
+            auth: None,
+            params: serde_json::to_value(params)
+                .expect("parameter serialization error - this should not happen"),
+        }
+    }
+
+    pub fn to_json(&self) -> JsonValue {
+        serde_json::to_value(self).expect("parameter serialization error - this should not happen")
+    }
+}
+
+/// Authentication uses the hashed user password in PHC format
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct ApiAuth(pub String);
```

Comment:
nit: if it's a hash, maybe we could use a hash type?

Reply:
It's not just a hash, but a string in the PHC format which is how the Argon API returns it

https://github.com/P-H-C/phc-string-format/blob/master/phc-sf-spec.md